fedimint-client-legacy/src/mint/mod.rs

```diff
@@ -521,14 +523,28 @@ impl MintClient {
     }
 
     pub async fn fetch_all_notes(&self) -> Vec<Result<OutPoint>> {
-        let active_issuances = &self.list_active_issuances().await;
+        let mut active_issuances = self.list_active_issuances().await;
         let mut results = vec![];
 
+        // Prevent trying to fetch way too many notes at the same time at the time.
+        // In normal cricumstances we should never hit this limit, but when using
+        // huge sums or hitting bugs, it is possible.
+        const LIMIT_NOTES_TO_CHECK: usize = 200;
+
+        // Shuffle issuances so we don't always retry the same ones, if there's a lot of
+        // them.
+        if LIMIT_NOTES_TO_CHECK < active_issuances.len() {
+            active_issuances.shuffle(&mut thread_rng());
+        }
+
+        let active_issuances = &active_issuances;
+
         #[cfg(not(target_family = "wasm"))]
         let mut futures = FuturesUnordered::<Pin<Box<dyn Future<Output = _> + Send>>>::new();
         #[cfg(target_family = "wasm")]
         let mut futures = FuturesUnordered::<Pin<Box<dyn Future<Output = _>>>>::new();
-        for (outpoint, _) in active_issuances {
+        // Try only limited number of issuances at the time
+        for (outpoint, _) in active_issuances.iter().take(LIMIT_NOTES_TO_CHECK) {
```

Comment:
Another way of limiting parallelism here, without limiting the number of operations, would be thorough the use of [`buffer_unordered`](https://docs.rs/futures/0.3.28/futures/stream/trait.StreamExt.html#method.buffer_unordered) (we'd first need to convert the active issuances into a stream I guess).